pkg/block/azure/multipart_block_writer.go

```diff
@@ -20,46 +23,58 @@ import (
 	}
 }
 
-func (m *MultipartBlockWriter) StageBlock(ctx context.Context, s string, seeker io.ReadSeeker, conditions azblob.LeaseAccessConditions, bytes []byte, options azblob.ClientProvidedKeyOptions) (*azblob.BlockBlobStageBlockResponse, error) {
-	return m.to.StageBlock(ctx, s, seeker, conditions, bytes, options)
+func (m *MultipartBlockWriter) StageBlock(ctx context.Context, base64BlockID string, body io.ReadSeekCloser, options *blockblob.StageBlockOptions) (blockblob.StageBlockResponse, error) {
+	return m.to.StageBlock(ctx, base64BlockID, body, options)
 }
 
-func (m *MultipartBlockWriter) CommitBlockList(ctx context.Context, ids []string, headers azblob.BlobHTTPHeaders, metadata azblob.Metadata, conditions azblob.BlobAccessConditions, tierType azblob.AccessTierType, tagsMap azblob.BlobTagsMap, options azblob.ClientProvidedKeyOptions) (*azblob.BlockBlobCommitBlockListResponse, error) {
+func (m *MultipartBlockWriter) CommitBlockList(ctx context.Context, ids []string, options *blockblob.CommitBlockListOptions) (blockblob.CommitBlockListResponse, error) {
 	m.etag = "\"" + hex.EncodeToString(m.reader.Md5.Sum(nil)) + "\""
 	base64Etag := base64.StdEncoding.EncodeToString([]byte(m.etag))
 
 	// write to blockIDs
 	pd := strings.Join(ids, "\n") + "\n"
-	_, err := m.toIDs.StageBlock(ctx, base64Etag, strings.NewReader(pd), conditions.LeaseAccessConditions, nil, options)
+	var leaseAccessConditions *blob.LeaseAccessConditions
+	if options.AccessConditions != nil {
+		leaseAccessConditions = options.AccessConditions.LeaseAccessConditions
+	}
+	_, err := m.toIDs.StageBlock(ctx, base64Etag, streaming.NopCloser(strings.NewReader(pd)), &blockblob.StageBlockOptions{
+		LeaseAccessConditions: leaseAccessConditions,
+	})
 	if err != nil {
-		return nil, fmt.Errorf("failed staging part data: %w", err)
+		return blockblob.CommitBlockListResponse{}, fmt.Errorf("failed staging part data: %w", err)
 	}
 	// write block sizes
 	sd := strconv.Itoa(int(m.reader.CopiedSize)) + "\n"
-	_, err = m.toSizes.StageBlock(ctx, base64Etag, strings.NewReader(sd), conditions.LeaseAccessConditions, nil, options)
+	_, err = m.toSizes.StageBlock(ctx, base64Etag, streaming.NopCloser(strings.NewReader(sd)), &blockblob.StageBlockOptions{
+		LeaseAccessConditions: leaseAccessConditions,
+	})
 	if err != nil {
-		return nil, fmt.Errorf("failed staging part data: %w", err)
+		return blockblob.CommitBlockListResponse{}, fmt.Errorf("failed staging part data: %w", err)
 	}
 
-	return &azblob.BlockBlobCommitBlockListResponse{}, err
+	return blockblob.CommitBlockListResponse{}, err
 }
 
-func completeMultipart(ctx context.Context, parts []block.MultipartPart, container azblob.ContainerURL, objName string, retryOptions azblob.RetryReaderOptions) (*block.CompleteMultiPartUploadResponse, error) {
+func (m *MultipartBlockWriter) Upload(_ context.Context, _ io.ReadSeekCloser, _ *blockblob.UploadOptions) (blockblob.UploadResponse, error) {
+	panic("Should not be called")
+}
```

Comment:
We needed to implement the interface - but we don't really need to use it.
I'm not sure what is the right implementation as I do not know how this will be used.